command/command.go

```diff
@@ -7,41 +7,50 @@ import (
 	"io"
 	"os"
 	"os/exec"
+	"strings"
 	"syscall"
-
-	log "github.com/Sirupsen/logrus"
 )
 
+// Status represents the current command status
+type Status string
+
+// Command statuses
 const (
-	statusIdle     = "idle"
-	statusRunning  = "running"
-	statusError    = "error"
-	statusUpdating = "updating"
-	statusDone     = "succeed"
-	statusStopped  = "stopped"
+	StatusIdle     Status = "idle"
+	StatusRunning  Status = "running"
+	StatusError    Status = "error"
+	StatusUpdating Status = "updating"
+	StatusDone     Status = "succeed"
+	StatusStopped  Status = "stopped"
 )
 
-// Command represents a os level command, which can also
-// receive a logger file in order to dump the output to it.
+// Command represents a os level command, which can also receive a logger file
+// in order to dump the output to it.
 type Command struct {
 	Log io.WriteCloser
 	Cmd *exec.Cmd
 
-	// stderr *bytes.Buffer // Maybe this will help us debugging when a command fails?
-
-	Slug    string
 	Version string
+	Updater Updater
```

Comment:
Ohh I see the problem now. The construtor is the problem.
What about drop `Updater` field and change `Updater` type from an interface to a function and receive an `Updater` in the Update method?

```go

type Updater func (*Command) (bool, error)

func (c *Command) Update(updater Updater) (bool, error)
```